### PR TITLE
add goreleaser and update release job

### PR DESCRIPTION
**Pull request by cpanato**
#### Summary
- add goreleaser and update release job

#### Ticket Link

Fixes: https://github.com/sigstore/helm-sigstore/issues/26

#### Release Note

```release-note
add goreleaser and update release job
```

### CODE REVIEW

**Review by sabre1041 — CHANGES_REQUESTED** [review]
This is great work. One small addition is required.

I tried testing on my own fork and ran into issues uploading the artifacts to the release

https://github.com/sabre1041/helm-sigstore/actions/runs/1635015585

Note: I had to make some additional changes due to the forked repository, but take a look to see if anything jumps out

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -1,69 +1,30 @@
...
-      - name: Upload release binaries
-        uses: svenstaro/upload-release-action@v2
-        with:
-          repo_token: ${{ secrets.GITHUB_TOKEN }}
-          file: dist/*
-          tag: ${{ github.ref }}
-          overwrite: true
-          file_glob: true
+      - name: release
+        run: make release
```
[the remark this answers is not in the document]
Comment (cpanato) [reply .github/workflows/release.yml:30]: thanks for the review, missed that

[the remark this answers is not in the document]
Comment (cpanato) [reply .github/workflows/release.yml:30]: found other issues in goreleaser that is fixed here https://github.com/goreleaser/goreleaser/pull/2798

[the remark this answers is not in the document]
Comment (sabre1041) [reply .github/workflows/release.yml:30]: Great fix upstream. Do we want to move forward without SBOM generation until its fixed upstream and then add it in once `v1.3.0`?

**Comment by cpanato** [comment]
@sabre1041 done
you can check the rehearsal here: https://github.com/cpanato/helm-sigstore/runs/4691160707?check_suite_focus=true
and the release here: https://github.com/cpanato/helm-sigstore/releases/tag/v99.99.99

**Review by sabre1041 — APPROVED** [review]
LGTM

**Comment by sabre1041** [comment]
Thanks @cpanato for you efforts around this task!

**Pull request merged**